x/sequencer/keeper/bond.go

```diff
@@ -41,17 +40,14 @@ func (k Keeper) TryUnbond(ctx sdk.Context, seq *types.Sequencer, amt sdk.Coin) e
 		return errorsmod.Wrap(err, "refund")
 	}
 	if seq.Tokens.IsZero() {
-		return errorsmod.Wrap(k.unbond(ctx, seq), "unbond")
+		k.unbond(ctx, seq)
 	}
+	k.SetSequencer(ctx, *seq)
```

Comment:
why add this?
The function is taking pointer to sequencer, it should be caller responsibility to set it

Reply:
this function does different stuff (e.g refund)
seems wrong to complete the function without storing the changes on the seq object
wdyt?
I see now that I have duplicate call to `k.SetSequencer(ctx, *seq)` by the caller as well